config/cmake/libhdf5.settings.cmake.in

```diff
@@ -79,7 +79,7 @@ Dimension scales w/ new references: @DIMENSION_SCALES_WITH_NEW_REF@
     With deprecated public symbols: @HDF5_ENABLE_DEPRECATED_SYMBOLS@
             I/O filters (external): @EXTERNAL_FILTERS@
                      Map (H5M) API: @H5_HAVE_MAP_API@
-                        Direct VFD: @H5_HAVE_DIRECT@
+                        Direct VFD: @HDF5_ENABLE_DIRECT_VFD@
```

Comment:
Even though you stop the configure if (HAVE_O_DIRECT AND HAVE_POSIX_MEMALIGN) fails, the more correct is the H5_HAVE_DIRECT setting, instead of the option setting.
Also it would be consistent with the other VFDs.

Reply:
If you want it to say ON/OFF instead of 1/0 (as everything else in this file does), you need to use HDF5_ENABLE_DIRECT_VFD.

Reply:
The other VFDs should be adjusted similarly.

libhdf5.settings is a mess in CMake and clearly handled as an afterthought.

Reply:
> If you want it to say ON/OFF instead of 1/0 (as everything else in this file does), you need to use HDF5_ENABLE_DIRECT_VFD.

Oh, yes we want yes/no in the file. As long as the option matches the actual setting, then the option is preferred.

Reply:
> The other VFDs should be adjusted similarly.
> 
> libhdf5.settings is a mess in CMake and clearly handled as an afterthought.

It was copied from the autotools version and tried to use similar variables.